File: labassignment3a/app/src/main/java/com/example/tipcalculator/MainActivity.java
package com.example.tipcalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import static java.lang.Float.parseFloat;

public class MainActivity extends AppCompatActivity {

    Button calculate,increase,decrease;
    EditText bill, percentageOf;
    TextView result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        calculate = findViewById(R.id.calculateButton);
        decrease = findViewById(R.id.decreasePercentage);
        increase = findViewById(R.id.increasePercentage);
        bill = findViewById(R.id.bill);
        percentageOf = findViewById(R.id.percentageOf);
        result = findViewById(R.id.result);

        calculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                float billF = parseFloat(bill.getText().toString());
                float percentageF = parseFloat(percentageOf.getText().toString());
                result.setText(percentageF + "% of " + billF + " is " + (percentageF / 100 * billF));
            }
        });
        increase.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                float billF = parseFloat(bill.getText().toString());
                float percentageF = parseFloat(percentageOf.getText().toString());
                result.setText(percentageF + "% increase in " + billF + " is " + ( billF + (percentageF / 100 * billF)));
            }
        });
        decrease.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                float billF = parseFloat(bill.getText().toString());
                float percentageF = parseFloat(percentageOf.getText().toString());
                result.setText(percentageF + "% decrease in " + billF + " is " + ( billF - (percentageF / 100 * billF)));
            }
        });

    }
}

File: labassignment3/app/src/main/java/com/example/tipcalculator/MainActivity.kt
package com.example.tipcalculator

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val calculateButton = findViewById<Button>(R.id.calculateButton)
        val increaseButton = findViewById<Button>(R.id.increasePercentage)
        val decreaseButton = findViewById<Button>(R.id.decreasePercentage)
        val percentageOf = findViewById<EditText>(R.id.percentageOf)
        val bill = findViewById<EditText>(R.id.bill)
        val total = findViewById<TextView>(R.id.result)
        calculateButton.setOnClickListener {
            val percentage = percentageOf.text.toString().toFloat()
            val totalBill = bill.text.toString().toFloat()
            total.text = percentage.toString() + "% of " +  totalBill.toString() + " is " + (percentage / 100 * totalBill).toString()
        }
        increaseButton.setOnClickListener{
            val percentage = percentageOf.text.toString().toFloat()
            val totalBill = bill.text.toString().toFloat()
            total.text = percentage.toString() + "% increase in " +  totalBill.toString() + " is " + ((percentage / 100 * totalBill) + totalBill)
        }
        decreaseButton.setOnClickListener{
            val percentage = percentageOf.text.toString().toFloat()
            val totalBill = bill.text.toString().toFloat()
            total.text = percentage.toString() + "% decrease in " +  totalBill.toString() + " is " +(totalBill - (percentage / 100 * totalBill))
        }
    }
}